src/interfaces/CachingMechanism.ts

```diff
@@ -20,5 +26,26 @@ export interface CachingMechanismInterface {
    * @returns True if the value was stored, otherwise false.
    * @throws {Error} If the value could not be stored.
    */
-  set<T>(key: string, value: T, ttl?: number, overrides?: unknown): Promise<boolean>;
+  set<ObjectType, OverrideType>(
+    key: string,
+    value: ObjectType,
+    ttl?: number,
+    overrides?: OverrideType
+  ): Promise<boolean>;
+
+  /**
+   * Identical to set, but returns the ID of the value stored. This is useful for storing values in a cache that
+   * are not known ahead of time. For example, if you want to store a value in a cache that is the result of a
+   * computation, you can use this method to store the value and retrieve the ID of the value stored.
+   * @param key The canonical key to store.
+   * @param value The value to store.
+   * @param ttl The time to live in seconds.
+   * @param overrides Any overrides to the default caching mechanism configuration for the given caching protocol.
+   */
+  setWithReturnID<ObjectType, OverrideType>(
```

Comment:
Do you think this should be in the interface? Or should we have `set` always return a `string` rather than `boolean`? Just curious what you think I'm unopinonated but keeping interfaces slim generally seems like a good idea

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/341/commits/aa633b002adab888f786e24d1a464b775e28e9c7